pkg/kubectl/cmd/set/set_env.go

```diff
@@ -405,25 +410,33 @@ func (o *EnvOptions) RunEnv(f cmdutil.Factory) error {
 
 	allErrs := []error{}
 
+	p, matched, err := o.PrintFlags.ToPrinter("env updated", o.DryRun)
+	if !matched {
+		return fmt.Errorf("unable to match a printer suitable for the output format %q", o.Output)
+	}
+	if err != nil {
+		return err
+	}
+
 	for _, patch := range patches {
 		info := patch.Info
 		if patch.Err != nil {
 			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
 			continue
 		}
 
-		// no changes
-		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
-			continue
-		}
-
 		if o.Local || o.DryRun {
-			if err := cmdutil.PrintObject(o.Cmd, patch.Info.AsVersioned(), o.Out); err != nil {
+			if err := p.PrintObj(patch.Info.AsVersioned(), o.Out); err != nil {
 				return err
 			}
 			continue
 		}
 
+		// no changes
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
```

Comment:
Moved this down here so that `--dry-run` prints a message every time.
Before, no output was printed if all patches were empty